pkg/malwaremanager/v1/clamav/open.go

```diff
@@ -0,0 +1,104 @@
+package malwaremanager
+
+import (
+	"node-agent/pkg/malwaremanager"
+	malwaremanager2 "node-agent/pkg/malwaremanager/v1/types"
+	"node-agent/pkg/utils"
+	"os"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/armosec/armoapi-go/armotypes"
+	"github.com/dustin/go-humanize"
+	"github.com/dutchcoders/go-clamd"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/open/types"
+	"github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+)
+
+func (c *ClamAVClient) handleOpenEvent(event *types.Event, containerPid uint32) malwaremanager.MalwareResult {
+	if event == nil {
+		return nil
+	}
+
+	// discard if it is an open for writing event
+	if slices.Contains(event.Flags, "O_WRONLY") {
```

Comment:
I think there is a possibility to check it with bit wise operations. It will be much more efficient.